### DISCUSSION TITLE
Flex Plugins Dashboard: Deleting a Plugin (spinner)

### DISCUSSION THREAD

**Discussion by Porlin** [twilio-labs/paste#1148] [Office hours] [answered]
The specific questions I have are...

- Looking for feedback on component usage - in particular the 'spinner' 
- Looking for feedback on the spinner interaction that turns into a 'tick' do you have other examples that use this pattern?

**Background:** Looking at a consistent way of using the spinner in a modal or inside a button when deleting a plugin inside Flex. 
Stage of design: High-fidelity
**Invision prototype:** https://twilio.invisionapp.com/share/XDZXU4B54NK

**Comment by serifluous** [comment, marked as answer]
Hey @Porlin ! I left some comments in the Invision link. Generally:

- We've seen a spinner used on the button rather than on the whole modal/screen for an action since you're loading (or disabling) the action rather than data on the modal.
- The "small spinner" direction seems visually clearest to me, but if it's important that the customer know the deletion status of each file, they seem to lose that information on the error state once the modal closes out. I left a note about the error path in general, but I wasn't sure if you were still exploring design solutions for it. Happy to talk about that more during office hours.

**Reply by Porlin** [reply]
Hey @serifluous Thank you for your responses. My teams prefer option two "small spinner" direction too so we can focus on that in design systems office hours. Also I will sync with Caroline to go over copy and we can work on a clearer error state message.
